### PR TITLE
Fixes #9586: Move techniques reference doc to the manual

### PR DESCRIPTION
**Pull request by amousset**
https://www.rudder-project.org/redmine/issues/9586

### CODE REVIEW

File: 8_reference/70_techniques.txt [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+
+These files must reside in a folder with a version number. For each Technique, there can be several versions, Rudder will let you choose the version when creating a Directive.
+
+===== Version number formating
+
+The version number follows a formating "a la Debian" as described here: http://www.debian.org/doc/debian-policy/ch-controlfields.html#s-f-Version, (without the debian_revision version)
+
+==== General Rules
+
+All the tag name are in upper case, all the attributes are in camel case:
```
Comment (ncharles) [inline 8_reference/70_techniques.txt]: "All the tag names in the .xml"

File: 8_reference/70_techniques.txt [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+
+<SECTION name="example" component="true" componentKey="variable_name">
+
+----
+
+==== Details of the metadata.xml file
+
+----
+
+<TECHNIQUE id="policy_unique_id" name="human_name_of_the_policy">
```
Comment (ncharles) [inline 8_reference/70_techniques.txt]: s/policy/technique/

File: 8_reference/70_techniques.txt [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+  <DESCRIPTION>Description of the policy</DESCRIPTION>
+  <LONG_DESCRIPTION>Long description of the policy</LONG_DESCRIPTION><br />  <DEPRECATED>Deprecation message</DEPRECATED>                <!-- Mark the Technique as deprecated, deprecation message is mandatory, Only available since Rudder 3.0 --><br />  <DISPLAY>true/false</DISPLAY>                               <!-- Define if the policy is displayed in the interface or not. Default value : true -->
```
Comment (ncharles) [inline 8_reference/70_techniques.txt]: could you have each tag on a separate line please ?

File: 8_reference/70_techniques.txt [hunk 4]
```diff
@@ -0,0 +1,309 @@
...
+  <COMPATIBLE>                                                <!-- Optional, describe the version of the OS and CFEngine Agent the policy has been tested on. Only for information purpose -->
+    <OS version=">=2.5">OS Name</OS>                          <!-- Optional; OS Name and version on which the Policy has been tested --> 
+    <AGENT version="<3.1">cfengine-community</AGENT>          <!-- Optional; Agent name and version on which the Policy has been tested -->
```
Comment (ncharles) [inline 8_reference/70_techniques.txt]: you can put 3.6 rather than 3.1

File: 8_reference/70_techniques.txt [hunk 5]
```diff
@@ -0,0 +1,309 @@
...
+  </COMPATIBLE>
+  <MULTIINSTANCE>true/false</MULTIINSTANCE>                   <!-- Optional; defines if several instances of this template with differents variables can be deployed on a node; default value : false -->
+  <SYSTEM>true/false</SYSTEM>                                 <!-- Optional, defines if this Policy Template is a system Policy Template (internal Rudder usage); default value : false -->
```
Comment (ncharles) [inline 8_reference/70_techniques.txt]: s/Policy Template/Technique/

File: 8_reference/70_techniques.txt [hunk 6]
```diff
@@ -0,0 +1,309 @@
...
+   <TML name="tmlName">                                       <!-- Container for a TML (without the trailing .st -->
+    <OUTPATH>relativ/path/of/file</OUTPATH>                   <!-- Optional; defines the relative path for the generated file for this template; default : policyName/version/tmlName.cf -->
+    <INCLUDED>true/false</INCLUDED>                           <!-- Optional; defines if the template must be in the inputs list of the generated promises; default : true -->
+   </TML>
+  </TMLS>
+  <FILES>                                                     <!-- List of files to be copied "as-is" with this Technique. StringTemplate parser is NOT used on these. -->
+    <FILE name="file.txt">                                    <!-- Container for a FILE. name (mandatory) = path to the file to copy, can be relative or absolute from RUDDER_CONFIGURATION_REPOSITORY/ (see below) --><br />    <FILE name="file2.txt"><OUTPATH>technique_name/newname.txt</OUTPATH></FILE> <br />    <FILE name="RUDDER_CONFIGURATION_REPOSITORY/directory/other/file.txt"><OUTPATH>technique_name/filename</OUTPATH></FILE><br /><span style='font-size: 11.1800003051758px;'> </FILES>
```
Comment (ncharles) [inline 8_reference/70_techniques.txt]: same here, can you have one tag per line ?

File: 8_reference/70_techniques.txt [hunk 7]
```diff
@@ -0,0 +1,309 @@
...
+----
+
+<ITEM>
+ <LABEL>Red</LABEL>
+ <VALUE>red</VALUE>
+</ITEM>
+
+----
+
+NOTE: INPUT fields in pre-2.6 needed to be manually escaped. Indeed, the CFEngine syntax doesn't allow quotes ( " ) within string, and the user had to manually escape them. Since version 2.6 of Rudder, the defaults INPUT fields are automatically escaped, meaning any quote will be written in the policies as \" ; and any backslash will be written as \\
```
Comment (ncharles) [inline 8_reference/70_techniques.txt]: I think we can drop the reference to 2.6, and say:
"NOTE: INPUT fields are automatically escaped, meaning any quote will be ...

File: 8_reference/70_techniques.txt [hunk 8]
```diff
@@ -0,0 +1,309 @@
...
+
+===== The <FILES> tag
+
+Example:
+
+----
+
+<FILES>
+<FILE name="file.txt"><OUTPATH>foo/bar/other-name.txt</OUTPATH></FILE>
+<FILE name="RUDDER_CONFIGURATION_REPOSITORY/some/absolute/file.txt"><OUTPATH>foo/bar/some-name.txt</OUTPATH></FILE>
```
Comment (ncharles) [inline 8_reference/70_techniques.txt]: when was that introduced ? Isn't it in 3.2 ?

File: 8_reference/70_techniques.txt [hunk 9]
```diff
@@ -0,0 +1,309 @@
...
+        <CONSTRAINT>
+        <UNIQUEVARIABLE>true</UNIQUEVARIABLE>
+    </INPUT>
+  </SECTIONS>
+
+----
+
+===== Password handling
+
+The password type allows to show an input text field whose content will be hashed when the form is submitted so that the password is never store in clear text. <img alt="rudder_password_field.jpg" height="168" src="%ATTACHURL%/rudder_password_field.jpg" width="945" />
```
Comment (amousset) [inline 8_reference/70_techniques.txt]: Missing image here, fixing.

File: 8_reference/72_techniques_templating.txt [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+The result would be:
+
+----
+
+"file[1][path]" string => "/var";
+"file[2][path]" string => "/bin";
+
+----
+
+===== Conditionnal display of a section
```
Comment (ncharles) [inline 8_reference/72_techniques_templating.txt]: s/display/writing/

**Comment by amousset** [comment]
Commit modified

File: 8_reference/70_techniques.txt [hunk 11]
```diff
@@ -0,0 +1,295 @@
...
+----
+
+<TECHNIQUE id="policy_unique_id" name="human_name_of_the_technique">
+  <DESCRIPTION>Description of the policy</DESCRIPTION>
+  <LONG_DESCRIPTION>Long description of the policy</LONG_DESCRIPTION>
+  <DEPRECATED>Deprecation message</DEPRECATED>                <!-- Mark the Technique as deprecated, deprecation message is mandatory, Only available since Rudder 3.0 -->
+  <DISPLAY>true/false</DISPLAY>                               <!-- Define if the policy is displayed in the interface or not. Default value : true -->
+  <COMPATIBLE>                                                <!-- Optional, describe the version of the OS and CFEngine Agent the policy has been tested on. Only for information purpose -->
+    <OS version=">=2.5">OS Name</OS>                          <!-- Optional; OS Name and version on which the Policy has been tested --> 
+    <AGENT version="<3.6">cfengine-community</AGENT>          <!-- Optional; Agent name and version on which the Policy has been tested -->
```
Comment (ncharles) [inline 8_reference/70_techniques.txt]: could you put >=3.6 ?

**Comment by amousset** [comment]
Commit modified

File: 8_reference/71_reports.txt [hunk 12]
```diff
@@ -0,0 +1,111 @@
...
+----
+
+@@Policy@@Type@@RuleId@@DirectiveId@@VersionId@@Component@@Key@@ExecutionTimeStamp##NodeId@#HumanReadableMessage
+
+----
+
+   * Policy : Human readable policy name
+   * Type : type of report (see bellow)
+   * RuleId : The Id of the Configuration Rule, autogenerated
+   * DirectiveId : The Id of the Policy Instance, autogenerated
```
Comment (ncharles) [inline 8_reference/71_reports.txt]: Id of the Directive

File: 8_reference/71_reports.txt [hunk 13]
```diff
@@ -0,0 +1,111 @@
...
+   * VersionId : the revision of the ConfigurationRule, autogenerated
+   * Component : the name of the component this policy is related to (if no component are defined in the policy.xml, then the policy name is used)
```
Comment (ncharles) [inline 8_reference/71_reports.txt]: s/policy/directive/

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
@ncharles PR updated, removed other occurrences of "policy".

**Comment by ncharles** [comment]
awesome, thank you !
can you correct the remaining typo, and self merge ?
Thank :+1:

**Comment by amousset** [comment]
Commit modified

**Pull request merged**